caffe2/core/nomnigraph/Converters/Caffe2.cc

```diff
@@ -13,44 +13,54 @@ getArgumentsFromOperator(caffe2::OperatorDef op) {
   return argMap;
 }
 
+nom::repr::NeuralNetOperator::NNLayout getLayout(std::map<std::string, caffe2::Argument> argMap) {
+  if (argMap.count("order")) {
+    auto order = argMap["order"].s();
+    if (order == "NCHW") {
+      return nom::repr::NeuralNetOperator::NNLayout::NCHW;
+    } else if (order == "NHWC") {
+      return nom::repr::NeuralNetOperator::NNLayout::NHWC;
+    }
+  }
+  return nom::repr::NeuralNetOperator::NNLayout::Undefined;
+}
+
+std::vector<int> getKernelShape(std::map<std::string, caffe2::Argument> argMap) {
+  // There are literally three ways to define shapes in Conv in Caffe2
+  std::vector<int> kernelShape;
+  if (argMap.count("kernel")) {
+    assert(argMap["kernel"].has_i() && "Invalid kernel argument");
+    int kernel = static_cast<int>(argMap["kernel"].i());
+    kernelShape = {kernel, kernel};
+  } else if (argMap.count("kernels")) {
+    for (auto i : argMap["kernels"].ints()) {
+      kernelShape.push_back(static_cast<int>(i));
+    }
+  } else if (argMap.count("kernel_h") && argMap.count("kernel_w")) {
+    assert(argMap["kernel_h"].has_i() && "Invalid kernel argument");
+    assert(argMap["kernel_w"].has_i() && "Invalid kernel argument");
+    int kernelH = static_cast<int>(argMap["kernel_h"].i());
+    int kernelW = static_cast<int>(argMap["kernel_w"].i());
+    kernelShape = {kernelH, kernelW};
+  } else {
+    assert(0 && "Could not parse kernel argument");
```

Comment:
This should be `CAFFE_THROW`. In Release mode (which is now default), `assert` expands to nothing.